### DISCUSSION TITLE
[general] tar_prune() equivalent for functions/objects no longer part of the pipeline

### DISCUSSION THREAD

**Discussion by benzipperer** [ropensci/targets#1312] [General]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

Is there a tar_prune() equivalent or recommended way to clean up the metadata for old objects and functions that are no longer part of the pipeline? 

I can use tar_invalidate() on specific names but I was wondering if there is a simpler way to clean things up without specifying the names of old objects or functions in the metadata, like tar_prune() does for targets.

Thank you for the awesome package!

**Comment by mattwarkentin** [comment]
Hi @benzipperer,

I believe `tar_prune()` already does what you're looking for, unless I am misunderstanding.
> Remove target values from _targets/objects/ and the cloud and remove target metadata from _targets/meta/meta for targets that are no longer part of the pipeline.

Link: https://docs.ropensci.org/targets/reference/tar_prune.html

**Reply by benzipperer** [reply]
I think tar_prune() just removes targets, which is great. I am wondering if there is a recommended similar way to remove from the metadata all old functions or global objects (as opposed to targets) that are no longer used, without specifying them explicitly through tar_invalidate().

For example, below old_function is still in the meta data after tar_prune().

``` r
library(targets)

tar_script({
  old_function = function(x) {1 + x}
  list(tar_target(y1, old_function(1)))
}, ask = FALSE)
tar_make()
#> ▶ dispatched target y1
#> ● completed target y1 [0 seconds]
#> ▶ ended pipeline [0.046 seconds]

tar_script({
  new_function = function(x) {2 + x}
  list(tar_target(y2, new_function(2)))
}, ask = FALSE)
tar_make()
#> ▶ dispatched target y2
#> ● completed target y2 [0 seconds]
#> ▶ ended pipeline [0.058 seconds]

tar_prune()
tar_meta()
#> # A tibble: 3 × 18
#>   name  type  data  command depend    seed path  time                size  bytes
#>   <chr> <chr> <chr> <chr>   <chr>    <int> <lis> <dttm>              <chr> <dbl>
#> 1 old_… func… e6cb… <NA>    <NA>   NA      <chr> NA                  <NA>     NA
#> 2 new_… func… 1606… <NA>    <NA>   NA      <chr> NA                  <NA>     NA
#> 3 y2    stem  20fe… 2e6a7f… 1797c… -1.50e9 <chr> 2024-08-01 12:35:38 948e…    51
#> # ℹ 8 more variables: format <chr>, repository <chr>, iteration <chr>,
#> #   parent <chr>, children <list>, seconds <dbl>, warnings <chr>, error <chr>
```

<sup>Created on 2024-08-01 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

**Reply by wlandau** [reply]
Sound like a reasonable change, please give 3c850abece5cb868c3959939ae3cf7cb0b616554 a try.

**Reply by benzipperer** [reply]
This is perfect, thank you!
